src/components/NewCommentForm.tsx

```
@@ -1,6 +1,107 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { FC, useState } from 'react';
+import { addPostComment } from './services/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  addComment: (comment: Comment) => void;
+  activePostId: number;
+};
+
+export const NewCommentForm: FC<Props> = ({
+  addComment = () => {},
+  activePostId,
+}) => {
+  const [inputName, setInputName] = useState('');
+  const [inputEmail, setInputEmail] = useState('');
+  const [inputText, setInputText] = useState('');
+
+  const [hasInputNameError, setHasInputNameError] = useState(false);
+  const [hasInputEmailError, setHasInputEmailError] = useState(false);
+  const [hasInputTextError, setHasInputTextError] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  function resetForm() {
+    setInputText('');
+  }
+
+  function clearForm() {
+    setInputText('');
+    setInputEmail('');
+    setInputName('');
+
+    setHasInputNameError(false);
+    setHasInputEmailError(false);
+    setHasInputTextError(false);
+  }
+
+  const handleSubmit = async (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+
+    if (inputName.length === 0) {
+      setHasInputNameError(true);
+    }
+
+    if (inputEmail.length === 0) {
+      setHasInputEmailError(true);
+    }
+
+    if (inputText.length === 0) {
+      setHasInputTextError(true);
+    }
+
+    if (
+      inputText.length === 0 ||
+      inputEmail.length === 0 ||
+      inputName.length === 0
+    ) {
+      return;
+    } else {
+      const message = {
+        name: inputName,
+        email: inputEmail,
+        body: inputText,
+      };
+
+      const newComment = { ...message, postId: activePostId };
+
+      setIsLoading(true);
+      addPostComment(newComment)
+        .then(commentar => {
+          addComment(commentar);
+          resetForm();
+        })
+        .catch(() => {})
```

The `.catch` block is currently empty. This means if the API call to add a comment fails, the error is swallowed, and the user receives no feedback. The loading spinner will just disappear, leaving the user to wonder if their comment was posted.

The task has an optional requirement: '(*) Handle `Add` and `Delete` errors so the user can retry'. You should implement some form of error notification here to inform the user about the failure.